apple-codesign/src/windows.rs

```diff
@@ -0,0 +1,736 @@
+    fn decrypt(&self, ciphertext: &[u8]) -> Result<Vec<u8>, RemoteSignError> {
+        // First, we need to check if the signer has a private key.
+        if self.hkey == NCRYPT_KEY_HANDLE::default() {
+            return Err(RemoteSignError::Crypto(
+                "certificate does not have a private key".into(),
+            ));
+        }
+
+        // We set the OAEP padding info.
+        let padding_info: *mut BCRYPT_OAEP_PADDING_INFO = &mut BCRYPT_OAEP_PADDING_INFO {
+            pszAlgId: BCRYPT_SHA256_ALGORITHM as *mut u16,
+            cbLabel: 0,
+            pbLabel: ptr::null_mut(),
+        };
+
+        // We decrypt using NCryptDecrypt.
+        let mut plaintext: Vec<u8> = Vec::new();
+        let mut plaintext_len: u32 = 0;
+        let result = unsafe {
+            NCryptDecrypt(
+                self.hkey,
+                ciphertext.as_ptr(),
+                ciphertext.len() as u32,
+                padding_info as *mut core::ffi::c_void,
+                ptr::null_mut(),
+                0,
+                &mut plaintext_len,
+                BCRYPT_PAD_OAEP,
+            )
+        };
+        if result != 0 {
+            return Err(RemoteSignError::Crypto(format!("error when attempting to decrypt ciphertext (NCryptDecrypt 1): 0x{:08X}", result)));
+        }
+        plaintext.resize(plaintext_len as usize, 0);
+        let result = unsafe {
+            NCryptDecrypt(
+                self.hkey,
+                ciphertext.as_ptr(),
+                ciphertext.len() as u32,
+                padding_info as *mut core::ffi::c_void,
+                plaintext.as_mut_ptr(),
+                plaintext.len() as u32,
+                &mut plaintext_len,
+                BCRYPT_PAD_OAEP,
+            )
+        };
+        if result != 0 {
+            return Err(RemoteSignError::Crypto(format!("error when attempting to decrypt ciphertext (NCryptDecrypt 2): 0x{:08X}", result)));
+        }
+        plaintext.resize(plaintext_len as usize, 0);
+
+        return Ok(plaintext);
+    }
+}
+
+impl PrivateKey for StoreCertificate {
+    fn as_key_info_signer(&self) -> &dyn KeyInfoSigner {
+        self
+    }
+
+    fn to_public_key_peer_decrypt(
+        &self,
+    ) -> Result<Box<dyn PublicKeyPeerDecrypt>, AppleCodesignError> {
+        Ok(Box::new(self.clone()))
+    }
+
+    fn finish(&self) -> Result<(), AppleCodesignError> {
+        Ok(())
+    }
+}
+
+impl StoreCertificate {
+    /// Obtain a new [CapturedX509Certificate] for this item.
+    pub fn as_captured_x509_certificate(&self) -> CapturedX509Certificate {
+        self.captured.clone()
+    }
+}
+
+fn find_certificates(
+    store_name: StoreName,
+    store_type: StoreType,
+) -> Result<Vec<StoreCertificate>, AppleCodesignError> {
+    let mut certs = vec![];
+
+    let store_type_as_str = <StoreType as std::convert::Into<&'static str>>::into(store_type);
+    let store_name_as_str = <StoreName as std::convert::Into<&'static str>>::into(store_name);
```

Comment:
Nit: shouldn't need the `std::convert::` prefix here since `Into` is in the global prelude.

Also, does `let store_type_str: &'static str = store_type.into()` not work?

Reply:
Thanks for the suggestion, should be fixed.